src/steam_network/websocket_client.py

```diff
@@ -127,8 +137,14 @@ async def run(self, create_future_factory: Callable[[], Future]=asyncio_future):
                 )
                 await sleep(RECONNECT_INTERVAL_SECONDS)
                 continue
+            except AuthenticationRequired:
+                logger.error("Authentication lost mid use. Restarting the socket, auth, and run loops")
```

Comment:
That whole function is bad legacy code we mostly left alone. 

That's there to handle a rare edge case: when the refresh token expires (or is manually revoked), the plugin used to silently fail. Now, when we catch that our login is no longer valid, we fire an error. This gets a little technical, apologies. It's less complex than it looks but the code is terrible.  

The cache is always stored in memory, but if also written to a db file for persistence. When we clear the cache, the code *immediately* loses the auth token. It's possible the file might not be updated if the plugin crashes or is closed before updating the file, but this is handled. 
When the plugin starts and is doing authentication, the first thing it checks is the user info cache. If it's invalid (in this example, the file was updated and use info was cleared), it switches to regular login. If it tries using the data in the cache and fails (in this example, the file was not updated and holds expired info), it switches to regular login as well. 

As for what this discotheques section does, it essentially resets the websocket to the state it'd be at when the plugin starts and you need to log in. In this state, it's waiting for authentication information from the main task. Once gog displays the login page and that info is obtained, we can go from there. 

This loop is tacked on to legacy code and I'm not a fan. But it's necessary to handle when the refresh tokens expire or are revoked manually

If this error arose while gog was asking for information, the code returns an AuthenticationRequired instead of what it asked for. If this occurs because steam sends us something server side, invalidating the cache has an added benefit of causing an AuthenticationRequired error the next time gog asks for something.

When gog gets AuthenticationRequired, it notifies the user "connection lost. Reconnect" and displays the login pa